Sources/Tracking/Extensions/DateExtensions.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+public extension Date {
+    // When decoding code to and from JSON, we lose milliseconds from the Date() objects.
+    // This can be difficult for unit testing because you might have to compare two
+    // identical Date() objects but one of the instances has the milliseconds removed.
+    // Use this to construct a new Date() object without the milliseconds to use in tests.
+    static var nowNoMilliseconds: Date {
+        let formatter = DateFormatter()
+        formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        let nowString = formatter.string(from: Date())
+
+        return formatter.date(from: nowString)!
```

Comment:
The app/SDK might crash if due of any reason, the return `formatter.date(from: nowString)!` is nil. Better to always safely unwrap the values. Looking at the use case, the chances are rare but always good to write safe code specially while unwrapping.